client/views/view_units.cpp

```diff
@@ -531,61 +531,65 @@ void get_units_view_data(struct unit_view_entry *entries,
 {
   *num_entries_used = 0;
 
-  if (!client_has_player()) {
-    return;
-  }
-
-  unit_type_iterate(unittype)
+  players_iterate(pplayer)
   {
-    int count = 0;           // Count of active unit type
-    int in_progress = 0;     // Count of being produced
-    int gold_cost = 0;       // Gold upkeep
-    int food_cost = 0;       // Food upkeep
-    int shield_cost = 0;     // Shield upkeep
-    bool upgradable = false; // Unit type is upgradable
-
-    unit_list_iterate(client_player()->units, punit)
+    if (client_has_player() && pplayer != client_player()) {
+      continue;
+    }
```

Comment:
How is this different from doing the following?
```cpp
auto pplayer = client_player();
if (!pplayer) {
  return;
}
```

Reply:
I didn't know that was an option I guess. Is the code better your way?

Reply:
I test it with your code and it breaks the view.